File: templates/problems/19_acm_io/03_sentinel_terminate/typescript/solution.ts
/**
 * Read pairs (a, b) until both are 0, return a+b for each pair (excluding sentinel).
 * Time: O(n), Space: O(n) for output.
 */
export function solveSentinelTerminate(inputData: string): string {
    const tokens = inputData.split(/\s+/).filter(t => t.length > 0);
    const outLines: string[] = [];
    for (let i = 0; i + 1 < tokens.length; i += 2) {
        const a = parseInt(tokens[i], 10);
        const b = parseInt(tokens[i + 1], 10);
        if (a === 0 && b === 0) break;
        outLines.push(String(a + b));
    }
    if (outLines.length === 0) return "";
    return outLines.join("\n") + "\n";
}

File: templates/problems/19_acm_io/03_sentinel_terminate/python/solution.py
def solve_sentinel_terminate(input_data: str) -> str:
    """Read pairs (a, b) until both are 0, return a+b for each pair (excluding sentinel).

    Time: O(n), Space: O(n) for output.
    """
    tokens = input_data.split()
    out_lines = []
    i = 0
    while i + 1 < len(tokens):
        a = int(tokens[i])
        b = int(tokens[i + 1])
        if a == 0 and b == 0:
            break
        out_lines.append(str(a + b))
        i += 2
    if not out_lines:
        return ""
    return "\n".join(out_lines) + "\n"
